services/wfm/src/main/java/org/openkilda/wfm/topology/event/OFELinkBolt.java

```diff
@@ -172,12 +216,20 @@ protected void doWork(Tuple tuple) {
         String json = tuple.getString(0);
         try {
             BaseMessage bm = MAPPER.readValue(json, BaseMessage.class);
+
             if (bm instanceof InfoMessage) {
                 InfoData data = ((InfoMessage)bm).getData();
-                if (data instanceof SwitchInfoData) {
+                if (data instanceof NetworkInfoData) {
+                    handleNetworkDump(tuple, (NetworkInfoData)data);
+                    isReceivedCacheInfo = true;
+                } else if (!isReceivedCacheInfo) {
+                    logger.debug("Bolt is not initialized mark tuple as fail");
```

Comment:
I think not because Storm is giving as guarantee all messages will proceed. 
> FailedMsgRetryManager is an interface which defines the retry strategy for a failed message. Default implementation is ExponentialBackoffMsgRetryManager which retries with exponential delays between consecutive retries. 

also code from org.apache.storm.kafka.SpoutConfig
```
// Failed message will be retried infinitely if retryLimit is less than zero. 
public int retryLimit = -1; 
```